### ISSUE TITLE
win32 support

### ISSUE THREAD

**Issue by ddatsh** [VolantMQ/volantmq#22]
win7 64 bit ok

set goarch=386
win7 32 bit

```
{"level":"info","logger":"mqtt.example","caller":"surgemq/surgemq.go:76","msg":"Starting application"}
{"level":"info","logger":"mqtt.example","caller":"surgemq/surgemq.go:82","msg":"Initializing configs"}
{"level":"info","logger":"mqtt.example","caller":"surgemq/surgemq.go:115","msg":"Started listener","id":"tcp4://:1883"}
panic: runtime error: invalid memory address or nil pointer dereference
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x40145b]

goroutine 7 [running]:
sync/atomic.StoreUint64(0x1146559c, 0x1, 0x0)
        c:/go/src/sync/atomic/asm_386.s:190 +0xb
github.com/troian/surgemq/buffer.(*Type).Close(0x11465560, 0x0, 0x0)
        d:/workspaces/go/sugermq/src/github.com/troian/surgemq/buffer/buffer.go:126 +0x3a
github.com/troian/surgemq/session.(*connection).stop(0x114980a0, 0x9f7a00)
        d:/workspaces/go/sugermq/src/github.com/troian/surgemq/session/connection.go:158 +0xbf
github.com/troian/surgemq/session.(*connection).onRoutineReturn(0x114980a0)
        d:/workspaces/go/sugermq/src/github.com/troian/surgemq/session/connection.go:315 +0x32
panic(0x7b0280, 0x9ee950)
        c:/go/src/runtime/panic.go:489 +0x22a
sync/atomic.LoadUint64(0x114655fc, 0xffffffff, 0x1)
        c:/go/src/sync/atomic/asm_386.s:159 +0xb
github.com/troian/surgemq/buffer.(*Type).isDone(0x114655c0, 0x114980d0)
        d:/workspaces/go/sugermq/src/github.com/troian/surgemq/buffer/buffer.go:630 +0x26
github.com/troian/surgemq/buffer.(*Type).WriteTo(0x114655c0, 0x11040a8, 0x114b9f70, 0x11040a8, 0x114b9f70, 0x7f1b01, 0x40a8d8)
        d:/workspaces/go/sugermq/src/github.com/troian/surgemq/buffer/buffer.go:195 +0x31
github.com/troian/surgemq/session.(*connection).sender(0x114980a0)
        d:/workspaces/go/sugermq/src/github.com/troian/surgemq/session/connection.go:304 +0x1e3
created by github.com/troian/surgemq/session.(*connection).start
        d:/workspaces/go/sugermq/src/github.com/troian/surgemq/session/connection.go:128 +0xd0
```

ref
https://my.oschina.net/u/115763/blog/198912

**Comment by troian** [comment]
@ddatsh I don't have any windows machine to check it quickly. Could you run on branch issue-22. I assume it might panic at some different place as there are possibly exists unalighed uint64 fields

**Comment by ddatsh** [comment]
seems to be fixed

**Comment by troian** [comment]
Merged PR #28
